### ISSUE TITLE
Focus Appearance (Minimum) needs more info on adjacent contrast

### ISSUE THREAD

**Issue by mbgower** [w3c/wcag#1929]
In 2.4.11 Focus Appearance (Minimum), the third bullet reads:

> Adjacent contrast: The contrasting area also has a contrast ratio of least 3:1 against adjacent colors in the focused component, or the contrasting area has a thickness of at least 2 CSS pixels.

The 2 CSS pixel thickness exception appears to let through a lot of poor designs -- until you bring in 1.4.11 Non-text Contrast and its wording covering "adjacent colors". The combination of the two SC's isn’t made clear enough, or demonstrated visually in the Understanding document.

Since 'adjacent colors" in 1.4.11 is only considered to apply to the background of the control ("colors adjacent to the component" in the Understanding document), whereas for 2.4.11's bullet it is talking about what is adjacent to the contrasting area, both the background and the component itself, it might make sense to make the normative language "against all adjacent colors"

Besides that possible normative change, the Understanding document needs revision. As it stands, when reading the  section titled ‘Relationship to Non-text Contrast’ I didn't really get the point. By this stage, a user has been reading this document for a long time, and I think the point of the section needs to be stated clearly and quickly right away. Frankly, I gave up. Reading it now, with my attention focused on "Adjacent contrast", it makes more sense, but I still find it hard to follow the visual example, etc. I think a clear and simple example that passes 1.4.11 but fails Focus Appearance (Minimum) would be useful. I also think an example that passes 2.4.11 but fails 1.4.11 would be useful, plus an example failing both and passing both.  I believe the key consideration for 1.4.11 is that it ensures the focus indicator meets 3:1 against the outer edge (the background) regardless of the stipulations of 2.4.11.

**Comment by mraccess77** [comment]
"Since 'adjacent colors" in 1.4.11 is only considered to apply to the background of the control ("colors adjacent to the component" in "  Hm.. Alastair has said that 1.4.11 applies to the adjacent background of the page and 2.4.11 to the components background.    We have some serious misunderstanding on SC 1.4.11 which has been around for 3 years and has been inconsistently tested in the field.  We need to get on the same page with SC 1.4.11 as without agreement our inter-rater reliability is low. 


Adjacent colors
For user interface components 'adjacent colors' means the colors adjacent to the component. For example, if an input has a white internal background, dark border, and white external background the 'adjacent color' to the component would be the white external background.

**Comment by patrickhlauke** [comment]
> We need to get on the same page with SC 1.4.11 as without agreement our inter-rater reliability is low

agree, this was also the outcome (or not) of that other discussion about use of color, non-text contrast, focus minimum ... https://github.com/w3c/wcag/issues/1775

**Comment by alastc** [comment]
> The 2 CSS pixel thickness exception appears to let through a lot of poor designs -- until you bring in 1.4.11 Non-text Contrast and its wording covering "adjacent colors". 

That was the underlying assumption: 1.4.11 covers the adjacent colors external to the component, so the new one shouldn't overlap.

That assumption is being raised again in the WCAG 2.x Misc items survey, should be next week.

**Comment by patrickhlauke** [comment]
> That was the underlying assumption: 1.4.11 covers the adjacent colors external to the component, so the new one shouldn't overlap.

https://github.com/w3c/wcag/issues/1775#issuecomment-867218125

Rather than the distinction of internal/external to the component, it makes more sense/is more logical (in my mind) to simply focus on "is it necessary to be able to distinguish this particular pair of adjacent colours yes or no?"

**Comment by mbgower** [comment]
> "is it necessary to be able to distinguish this particular pair of adjacent colours yes or no?"

Yes, Patrick, that is the crux of the issue. So the scenario, as i see it, is that 1.4.11 established that a component, regardless of its state, must contrast 3:1 against its adjacent colours -- which in the sense of a complete component is relatively easier to understand as the background on which the component sits.

However, when we begin talking about the focus indicator area, we are now talking not just about the contrast with the background on which the component sits, but also the contrast between the focus indicator area and the parts of the component itself that are adjacent to that indicator.

So one way of tackling this which _may_ be clearer for 2.4.11, is to **only** focus on the parts of the component that abut (AKA are adjacent to) the focus indicator area. Then we leave 1.4.11 to handle the background contrast.

**Comment by patrickhlauke** [comment]
what if the outline/new border has exactly the same colour as the background of the control, effectively making it grow? then its contrast would be zero, but visually it's still discernible (depending how thick it grows).

**Comment by mbgower** [comment]
I'm not following you. If the new border/focus indicator is exactly the same colour as the background, it would be an invisible indicator and fail.

**Comment by mbgower** [comment]
Or are you suggesting the outside 2 px perimeter of the button would change colour to the background colour, effectively making the button appear to _shrink_?

That would be an odd indicator of focus, but in that scenario what I'm talking about ( measuring the abutting parts of control to indicator) would still apply. 2.4.11 would ensure that the focus indicator area contrasts with the control itself; 1.4.11 would ensure that the control's perimeter (which in this scenario wouldn't include the focus indicator, since it's not perceivable') also contrasts with the background.

i think I see your point, though. If the adjacent bullet does not apply in all directions, then in your scenario, the 2 CSS pixel thickness would not be invoked.

**Comment by patrickhlauke** [comment]
Figure 15 in https://www.w3.org/WAI/WCAG22/Understanding/focus-appearance-minimum.html

![three buttons, center one focused, appearing thicker](https://user-images.githubusercontent.com/895831/123297994-fbc2b780-d50f-11eb-9508-1705206613b2.png)

**Comment by mbgower** [comment]
Oh, when you said 'background' you didn't mean it in the sense of page background. LOL, okay, another way language is failing this one. Yeah, so same argument -- we HAVE to apply the adjacent assessment to _all_ adjacents or we lose the 2CSS requirement.

**Comment by alastc** [comment]
Based on the changes in #2028, the SC text for the "Adjacent contrast" bullet is now:
> Adjacent contrast: If the contrasting area does not have a contrast ratio of least 3:1 against adjacent colors in the focused component then the area has a thickness of at least 2 CSS pixels.

I've just noticed that wiped out a previous adjustment to include "contributing area".

How about:
> Adjacent contrast: If the contrasting area does not have a contrast ratio of least 3:1 against adjacent colors in the focused component, then the contributing area has a thickness of at least 2 CSS pixels.

**Comment by alastc** [comment]
> Since 'adjacent colors" in 1.4.11 is only considered to apply to the background of the control ("colors adjacent to the component" in the Understanding document), whereas for 2.4.11's bullet it is talking about what is adjacent to the contrasting area, both the background and the component itself, it might make sense to make the normative language "against all adjacent colors"

If you add something around/outside the control, the background the control is on is the outside adjacent colours. In that scenario, the outside adjacent colours = change of contrast.

The scenario that bullet is trying to catch is where you add a thin line around a control that passes change of contrast, but blends into the control. Therefore we don't need to say there is a requirement for contrast with the colours outside the control, although I guess it doesn't do any harm.

> I think a clear and simple example that passes 1.4.11 but fails Focus Appearance (Minimum) would be useful. I also think an example that passes 2.4.11 but fails 1.4.11 would be useful, plus an example failing both and passing both. 

Referencing the [new examples in non-text contrast](https://w3c.github.io/wcag/understanding/non-text-contrast.html#figure-focus-inner), this would pass 1.4.11, but fail 2.4.11:
![image](https://user-images.githubusercontent.com/216630/135311530-618beb60-ff20-4776-bab4-82e5e1eb51e6.png) 
(1px dark outline around a dark button.)

As would:
![image](https://user-images.githubusercontent.com/216630/135311997-dccf6345-656f-4f65-aade-4737394a0214.png) 
(Yellow 1px box line that is partially in and partially out of the dark button.)

I can't think of one that would pass 2.4.11 but fail 1.4.11. The requirement is stricter, particularly using 'change of contrast' might mean you can't do it that way around.

**Comment by mraccess77** [comment]
"Since 'adjacent colors" in 1.4.11 is only considered to apply to the background of the control ("colors adjacent to the component" in the Understanding document), " 

This is the point we have been discussing lately and the fact is that it depends.  Our latest example show that in some cases the adjacent colors are to the page background but some are to the control itself and some are both in specific cases (when the border is used to communicate focus).

**Comment by alastc** [comment]
I've tried to simplify the 'relation to NTC' section in #2060.
